src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.service.operation.OperationHandler;
+import core.basesyntax.storage.Storage;
+import core.basesyntax.strategy.OperationStrategy;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class ShopServiceImpl implements ShopService {
+    private final Map<String, Integer> storage = Storage.getStorage();
```

Comment:
It's not recommended to use the storage map directly in the service layer. Instead, you should use dependency injection to pass the storage to the operation handlers. This will make your code more in line with the SOLID principles, specifically the Dependency Inversion Principle.